sql/core/src/test/java/test/org/apache/spark/sql/JavaDatasetSuite.java

```diff
@@ -302,16 +300,23 @@ public void testSetOperation() {
 
     Dataset<String> unioned = ds.union(ds2);
     Assert.assertEquals(
-      Arrays.asList("abc", "abc", "foo", "foo", "xyz", "xyz"),
-      sort(unioned.collectAsList().toArray(new String[0])));
+      asSet("abc", "abc", "foo", "foo", "xyz", "xyz"),
```

Comment:
Why did you change this? The previous version is ordering tolerant no? This changes the test a bit wrt duplicates.